recipes-bsp/secure-boot-otp-provisioning/README.md

```diff
@@ -0,0 +1,149 @@
+# IOT2050 secure boot customer key provisioning
+
+IOT2050 Advanced PG2 enables the SoC extend OTP area for storing the customer
+key used for signing the firmware & software that validated by SEBoot and afterwards.
+
+Below are the steps to programming keys into this OTP area.
+
+## Create the keys
+
+First, you need to generate at least two sets of key pair under the folder
+`files/keys`. At most you can have three key pair sets. An example of key pairs
+are already there for reference.
+
+These keys are named `MPK`, `SMPK` and `BMPK` respectively.
+
+```shell
+openssl req -x509 -newkey rsa:4096 -keyout custMpk.pem -nodes -outform pem -out custMpk.crt -sha256
+openssl req -x509 -newkey rsa:4096 -keyout custSmpk.pem -nodes -outform pem -out custSmpk.crt -sha256
+# optional the third key
+#openssl req -x509 -newkey rsa:4096 -keyout custBmpk.pem -nodes -outform pem -out custBmpk.crt -sha256
+```
+
+> :warning:
+> Do remove the example dummy keys under the keys folder!
+
+## Building the signed firmware image containing the key provisioning data
+
+The next step is to build the signed firmware image together with the key
+provisioning data.
+
+Before start the build, copy the currently using key set is copied to
+`recipes-bsp/u-boot/files/keys/custMpk.pem`.
+
+```shell
+./kas-container build kas-iot2050-boot-pg2.yml:kas/opt/secure-boot.yml:kas/opt/key-provision.yml
+```
+
+> Warning: The default key sets within kas/opt/key-provision.yml only contain
+> MPK and SMPK. To include BMPK as well, please refer to
+[Flexible key provisioning](#flexible-key-provisioning).
+
+## Update the firmware to provisioning the keys
+
+Copy the built firmware image to IOT2050 and run the firmware update against the
+new firmware image:
+
+```shell
+iot2050-firmware-update -f iot2050-pg2-image-boot.bin
+```
+
+Reboot the IOT2050 and follow the instruction on the device. The debug serial
+port has to be connected to give your consent to the provisioning routine.
+
+> Warning: after provisioning, all application must be signed with the currently
+> using key.
+
+The key provisioning just need be run once. Later secure boot firmware building
+should drop the `kas/opt/key-provision.yml`
```

Comment:
While the use-case of first-boot provisioning makes sense, I'm rather leaning towards installing that provisioning data outside of the u-boot build.

Reply:
Either way looks OK to me, I just want to keep the same manner as what we've done for WDT.

Or do you foresee this is a upstream no-go?

Reply:
On the other side, a little different with wdt:

Once wdt is required, then the wdt will be included always. However this otpcmd data is only needed once, but will always produce the warning of missing-blob, that could be annoying.

Reply:
Exactly.